fedora_messaging/twisted/protocol.py

```diff
@@ -0,0 +1,296 @@
+# This file is part of fedora_messaging.
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from __future__ import absolute_import, unicode_literals
+
+import logging
+
+import pika
+import pkg_resources
+from pika.adapters.twisted_connection import TwistedProtocolConnection
+from twisted.internet import defer, error
+# twisted.logger is available with Twisted 15+
+from twisted.python import log
+
+from .._session import get_message, get_serialized_message
+from ..exceptions import Nack, Drop, HaltConsumer, ValidationError
+
+
+_pika_version = pkg_resources.get_distribution('pika').parsed_version
+
+
+class FedoraMessagingProtocol(TwistedProtocolConnection):
+    """A Twisted Protocol for the Fedora Messaging system.
+
+    This protocol builds on the generic pika AMQP protocol to add calls
+    specific to the Fedora Messaging implementation.
+    """
+
+    name = 'FedoraMessaging:Protocol'
+
+    def __init__(self, parameters, confirms=False):
```

Comment:
Since this isn't exposed as a public API it shouldn't matter, but it seems like confirms=True is best since it's safer and probably the option you want unless you know what you're doing.

Reply:
Yeah, I set it to False by default because it wasn't implemented in pika's Twisted integration anyway. My pika PR now contains it, but it's still unreleased. Also it's a RabbitMQ extension IIRC.